From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Todd Kjos <tkjos@google.com>
Date: Wed, 4 Dec 2019 12:03:54 -0800
Subject: ANDROID: ASoC: Fix 'allmodconfig' build break

Export 'arch_setup_dma_ops' to fix the dependency added
if CONFIG_SND_SOC=m:

  ERROR: "arch_setup_dma_ops" [sound/soc/snd-soc-core.ko] undefined!

This should be followed up with a refactor that removes the dependency
(tracked in the cited bug).

Bug: 144369166
Fixes: 9cb7ec3e9be4 ("ANDROID: ASoC: core - add hostless DAI support")
Change-Id: Icf2daf1d92597e6dc9e723799ed7cea73e961227
Signed-off-by: Todd Kjos <tkjos@google.com>
---
 arch/arc/mm/dma.c               | 1 +
 arch/arm/mm/dma-mapping-nommu.c | 1 +
 arch/arm/mm/dma-mapping.c       | 1 +
 arch/arm64/mm/dma-mapping.c     | 1 +
 arch/mips/mm/dma-noncoherent.c  | 1 +
 5 files changed, 5 insertions(+)

diff --git a/arch/arc/mm/dma.c b/arch/arc/mm/dma.c
--- a/arch/arc/mm/dma.c
+++ b/arch/arc/mm/dma.c
@@ -104,3 +104,4 @@ void arch_setup_dma_ops(struct device *dev, u64 dma_base, u64 size,
 	dev_info(dev, "use %scoherent DMA ops\n",
 		 dev->dma_coherent ? "" : "non");
 }
+EXPORT_SYMBOL_GPL(arch_setup_dma_ops);
diff --git a/arch/arm/mm/dma-mapping-nommu.c b/arch/arm/mm/dma-mapping-nommu.c
--- a/arch/arm/mm/dma-mapping-nommu.c
+++ b/arch/arm/mm/dma-mapping-nommu.c
@@ -209,3 +209,4 @@ void arch_setup_dma_ops(struct device *dev, u64 dma_base, u64 size,
 	if (!dev->archdata.dma_coherent)
 		set_dma_ops(dev, &arm_nommu_dma_ops);
 }
+EXPORT_SYMBOL_GPL(arch_setup_dma_ops);
diff --git a/arch/arm/mm/dma-mapping.c b/arch/arm/mm/dma-mapping.c
--- a/arch/arm/mm/dma-mapping.c
+++ b/arch/arm/mm/dma-mapping.c
@@ -2324,6 +2324,7 @@ void arch_setup_dma_ops(struct device *dev, u64 dma_base, u64 size,
 #endif
 	dev->archdata.dma_ops_setup = true;
 }
+EXPORT_SYMBOL_GPL(arch_setup_dma_ops);
 
 void arch_teardown_dma_ops(struct device *dev)
 {
diff --git a/arch/arm64/mm/dma-mapping.c b/arch/arm64/mm/dma-mapping.c
--- a/arch/arm64/mm/dma-mapping.c
+++ b/arch/arm64/mm/dma-mapping.c
@@ -57,3 +57,4 @@ void arch_setup_dma_ops(struct device *dev, u64 dma_base, u64 size,
 		dev->dma_ops = &xen_swiotlb_dma_ops;
 #endif
 }
+EXPORT_SYMBOL_GPL(arch_setup_dma_ops);
diff --git a/arch/mips/mm/dma-noncoherent.c b/arch/mips/mm/dma-noncoherent.c
--- a/arch/mips/mm/dma-noncoherent.c
+++ b/arch/mips/mm/dma-noncoherent.c
@@ -141,4 +141,5 @@ void arch_setup_dma_ops(struct device *dev, u64 dma_base, u64 size,
 {
 	dev->dma_coherent = coherent;
 }
+EXPORT_SYMBOL_GPL(arch_setup_dma_ops);
 #endif
